### ISSUE TITLE
 "Methods and properties that don't access instance data should be static" on an interface implementation

### ISSUE THREAD

**Issue by alexeyzimarev** [SonarSource/sonar-csharp#1068]
We are getting some false positives on "Class methods and properties that don't access instance data can be static to prevent any misunderstanding about the contract of the method."

The rule is fine but it is not applicable for interface implementation members. If I follow the suggestion and "fix" the issue, the code will not compile because the class will not implement the interface.

**Comment by Evangelink** [comment]
Hi @alexeyzimarev,

Would you mind sharing an example? If I try to test with the following example I can't get any issue reported.

```
public interface IFoo
{
    void DoSomething();
}

public class Foo : IFoo
{
    public void DoSomething()
    {
        var x = 5;
    }
}
```

But if I remove the `: IFoo` I will do have the issue.

**Comment by alexeyzimarev** [comment]
I have issue in both cases but it can be the old version again. I will have to check how to get Sonar C# updated before giving any more feedback.
